extension.js

```diff
@@ -36,6 +36,10 @@ class ExtensionDBusInterface {
             return;
 
         Main.wm.skipNextEffect(current_window.get_compositor_private());
+        if (settings.get_boolean('window-maximize')) {
```

Comment:
IMO this `if` is unnecessary - if the execution reached this point, the window is maximized (look at the `if` at the beginning of the function)

Reply:
But the fact that the window is maximized vertically is different from whether the user has or not pressed the "maximize window" button, isn't it?